packages/opentelemetry-resources/src/detectors/NoopDetector.ts

```diff
@@ -14,13 +14,13 @@
  * limitations under the License.
  */
 
-import { Detector } from '../types';
-import { IResource } from '../IResource';
-import { noopDetectorSync } from './NoopDetectorSync';
+import { ResourceDetector } from '../types';
 
-export class NoopDetector implements Detector {
-  detect(): Promise<IResource> {
-    return Promise.resolve(noopDetectorSync.detect());
+export class NoopDetector implements ResourceDetector {
+  detect() {
```

Comment:
`: DetectedResource` ...if we want to be consistent about public interface return types, I'd leave a suggestion but this requires adding the import too